homeassistant/components/solarlog/const.py

```diff
@@ -17,83 +30,187 @@
+    SolarLogSensorEntityDescription(
+        key="yield_day",
+        json_key="yieldDAY",
+        name="yield day",
+        icon="mdi:solar-power",
+        native_unit_of_measurement=ENERGY_KILO_WATT_HOUR,
+    ),
+    SolarLogSensorEntityDescription(
+        key="yield_yesterday",
+        json_key="yieldYESTERDAY",
+        name="yield yesterday",
+        icon="mdi:solar-power",
+        native_unit_of_measurement=ENERGY_KILO_WATT_HOUR,
+    ),
+    SolarLogSensorEntityDescription(
+        key="yield_month",
+        json_key="yieldMONTH",
+        name="yield month",
+        icon="mdi:solar-power",
+        native_unit_of_measurement=ENERGY_KILO_WATT_HOUR,
+    ),
+    SolarLogSensorEntityDescription(
+        key="yield_year",
+        json_key="yieldYEAR",
+        name="yield year",
+        icon="mdi:solar-power",
+        native_unit_of_measurement=ENERGY_KILO_WATT_HOUR,
+    ),
+    SolarLogSensorEntityDescription(
+        key="yield_total",
+        json_key="yieldTOTAL",
+        name="yield total",
+        icon="mdi:solar-power",
+        native_unit_of_measurement=ENERGY_KILO_WATT_HOUR,
+        state_class=STATE_CLASS_TOTAL_INCREASING,
+    ),
+    SolarLogSensorEntityDescription(
+        key="consumption_ac",
+        json_key="consumptionAC",
+        name="consumption AC",
+        native_unit_of_measurement=POWER_WATT,
+        device_class=DEVICE_CLASS_POWER,
+        state_class=STATE_CLASS_MEASUREMENT,
+    ),
+    SolarLogSensorEntityDescription(
+        key="consumption_day",
+        json_key="consumptionDAY",
+        name="consumption day",
+        native_unit_of_measurement=ENERGY_KILO_WATT_HOUR,
+        device_class=DEVICE_CLASS_ENERGY,
+    ),
+    SolarLogSensorEntityDescription(
+        key="consumption_yesterday",
+        json_key="consumptionYESTERDAY",
+        name="consumption yesterday",
+        native_unit_of_measurement=ENERGY_KILO_WATT_HOUR,
+        device_class=DEVICE_CLASS_ENERGY,
+    ),
+    SolarLogSensorEntityDescription(
+        key="consumption_month",
+        json_key="consumptionMONTH",
+        name="consumption month",
+        native_unit_of_measurement=ENERGY_KILO_WATT_HOUR,
+        device_class=DEVICE_CLASS_ENERGY,
+    ),
+    SolarLogSensorEntityDescription(
+        key="consumption_year",
+        json_key="consumptionYEAR",
+        name="consumption year",
+        native_unit_of_measurement=ENERGY_KILO_WATT_HOUR,
+        device_class=DEVICE_CLASS_ENERGY,
+    ),
+    SolarLogSensorEntityDescription(
+        key="consumption_total",
+        json_key="consumptionTOTAL",
+        name="consumption total",
+        native_unit_of_measurement=ENERGY_KILO_WATT_HOUR,
+        device_class=DEVICE_CLASS_ENERGY,
+        state_class=STATE_CLASS_TOTAL_INCREASING,
+    ),
+    SolarLogSensorEntityDescription(
+        key="total_power",
+        json_key="totalPOWER",
+        name="total power",
+        icon="mdi:solar-power",
+        native_unit_of_measurement="Wp",
```

Comment:
I don't think this is the correct unit of measurement. I'm assuming this means Watt-peak, but that's not a scientific unit of measurement but what kind of power measurement this is. The unit should just be Watt (POWER_WATT) and we should specify that it's peak power in the name.

https://en.wikipedia.org/wiki/Nominal_power_(photovoltaic)#Watt-peak

Reply:
I think we should stick with scientific standards as far as possible.

Reply:
Added a new PR with this change in #55110. Will update the docs tomorrow.